scripts/github_issue_retriever.py

```diff
@@ -7,98 +7,111 @@
+        self.github = Github(
+            owner=owner, repo=repo, state=state, retrieve_events=retrieve_events
+        )
+
+    def retrieve_issues(self) -> None:
 
         last_modified = None
-        db.download(github.GITHUB_ISSUES_DB)
+        db.download(self.github.db_path)
 
         try:
-            last_modified = db.last_modified(github.GITHUB_ISSUES_DB)
+            last_modified = db.last_modified(self.github.db_path)
         except Exception:
             pass
 
         if last_modified:
             logger.info(
                 f"Retrieving issues modified or created since the last run on {last_modified.isoformat()}"
             )
-            data = github.fetch_issues_updated_since_timestamp(
-                owner, repo, state, last_modified.isoformat(), retrieve_events
+            data = self.github.fetch_issues_updated_since_timestamp(
+                last_modified.isoformat()
             )
 
-            if retrieve_private:
+            if self.retrieve_private:
                 logger.info(
                     "Replacing contents of auto closed public issues with private issues content"
                 )
-                replace_with_private(data)
+                self.replace_with_private(data)
 
             updated_ids = set(issue["id"] for issue in data)
 
             logger.info(
                 "Deleting issues that were changed since the last run and saving updates"
             )
-            github.delete_issues(lambda issue: issue["id"] in updated_ids)
+            self.github.delete_issues(lambda issue: issue["id"] in updated_ids)
 
-            db.append(github.GITHUB_ISSUES_DB, data)
+            db.append(self.github.db_path, data)
             logger.info("Updating finished")
         else:
             logger.info("Retrieving all issues since last_modified is not available")
-            github.download_issues(owner, repo, state, retrieve_events)
+            self.github.download_issues()
 
-            if retrieve_private:
+            if self.retrieve_private:
                 logger.info(
                     "Replacing contents of auto closed public issues with private issues content"
                 )
 
-                all_issues = list(github.get_issues())
-                updated_issues, updated_ids = replace_with_private(all_issues)
+                all_issues = list(self.github.get_issues())
+                updated_issues, updated_ids = self.replace_with_private(all_issues)
 
                 logger.info(
                     "Deleting public issues that were updated and saving updates"
                 )
-                github.delete_issues(lambda issue: issue["id"] in updated_ids)
-                db.append(github.GITHUB_ISSUES_DB, updated_issues)
+                self.github.delete_issues(lambda issue: issue["id"] in updated_ids)
+                db.append(self.github.db_path, updated_issues)
+
+        zstd_compress(self.github.db_path)
+
+    def replace_with_private(
```

Comment:
Could you keep this function where it was, to minimize the diff?

Reply:
I've moved it inside the Retriever class because it's using its github instance in `self.github.fetch_issue_by_number`